substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java

```diff
@@ -0,0 +1,152 @@
+package com.oracle.svm.core.jdk.jfr.remote;
+
+import com.oracle.svm.core.jdk.jfr.JfrOptions;
+import com.oracle.svm.core.jdk.jfr.logging.JfrLogger;
+
+import jdk.jfr.Configuration;
+import jdk.jfr.Recording;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.text.ParseException;
+
+/*
+ * Handlers the startup and teardown of JFR when run from either the command line or remote interface.
+ * Implemented by standard JFR calls as user code might do.
+ */
+
+public class JfrAutoSessionManager {
+
+    private static final String DEFAULT_JFR_FILENAME = "recording.jfr";
+    private static final String DEFAULT_JFR_CONFIG = "default";
+
+    Configuration jfrConfiguration;
+    String filename;
+    Recording recording;
+    static JfrAutoSessionManager instance;
+
+    static void out(String msg) {
+        JfrLogger.logInfo("JFR.AutoStart " + msg);
+    }
+
+    JfrAutoSessionManager() throws IOException, ParseException {
+        this(DEFAULT_JFR_CONFIG);
+    }
+
+    JfrAutoSessionManager(String configName) throws IOException, ParseException {
+        out("JfrAutoSessionManager(" + configName + ")");
+        jfrConfiguration = Configuration.getConfiguration(configName);
+    }
+
+    /*
+     *   JFR-TODO if these timers expire after the main program has ended, the process hangs.
+     */
+    void startSession(String fn, long delayMilliseconds, long durationMilliseconds) {
+        if (delayMilliseconds > 0) {
+            out("scheduling JFR start in " + delayMilliseconds + "ms.");
+            new java.util.Timer().schedule(
+                    new java.util.TimerTask() {
+                        @Override
+                        public void run() {
+                            startSession(fn, 0, durationMilliseconds);
+                        }
+                    },
+                    delayMilliseconds
+            );
+        } else {
+            out("starting JFR session(" + fn + ")");
+            filename = fn;
+            recording = new Recording(jfrConfiguration);
+            recording.start();
+            try {
+                recording.setDestination(Paths.get(filename));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            if (durationMilliseconds > 0) {
+                out("scheduling JFR stop in " + durationMilliseconds + "ms.");
+                new java.util.Timer().schedule(
+                        new java.util.TimerTask() {
+                            @Override
+                            public void run() {
+                                stopSession(true, false);
+                            }
+                        },
+                        durationMilliseconds
+                );
+            }
+        }
+    }
+
+    void stopSession(boolean write, boolean clear) {
```

Comment:
The boolean write is only used in log output and the current callers always supply true. What is its purpose?

Reply:
unimplemented (yet) flexibility in the remote control code.

Reply:
Okay, that is fine with me.